Library/Homebrew/cmd/pull.rb

```diff
@@ -88,23 +71,34 @@ def pull
       revision = `git rev-parse --short HEAD`.strip
       branch = `git symbolic-ref --short HEAD`.strip
 
-      unless branch == "master"
-        opoo "Current branch is #{branch}: do you need to pull inside master?" unless ARGV.include? "--clean"
+      unless branch == "master" || ARGV.include?("--clean") || ARGV.include?("--branch-okay")
+        opoo "Current branch is #{branch}: do you need to pull inside master?"
       end
 
-      pull_url url
+      patch_puller = PatchPuller.new(url)
+      patch_puller.fetch_patch
+      patch_changes = files_changed_in_patch(patch_puller.patchpath)
+      is_bumpable = patch_changes[:formulae].length == 1 && patch_changes[:others].empty?
+      if do_bump
+        odie "No changed formulae found to bump" if patch_changes[:formulae].empty?
+        odie "Can only bump one changed formula" unless patch_changes[:formulae].length == 1
+        odie "Can not bump if non-formula files are changed" unless patch_changes[:others].empty?
+      end
+      if is_bumpable
+        old_versions = current_versions_from_info_external(patch_changes[:formulae].first)
+      end
+      patch_puller.apply_patch
 
       changed_formulae = []
 
       Utils.popen_read(
-        "git", "diff-tree", "-r", "--name-only",
-        "--diff-filter=AM", revision, "HEAD", "--", tap.formula_dir.to_s
+          "git", "diff-tree", "-r", "--name-only",
+          "--diff-filter=AM", revision, "HEAD", "--", tap.formula_dir.to_s
```

Comment:
Yeah. This was a result of me using the "auto-indent" in RubyMine. It defaults to a four-space "continuation indent" inside multiline `(...)` and `{...}`. Fixed.